.github/workflows/release.yml

```diff
@@ -83,111 +109,159 @@ jobs:
+            image_tag: latest
+            dockerfile: dockerfile/kpow/Dockerfile
+            script: build_all.sh
+          - image_id: factorhouse/kpow
+            image_tag: ${{github.event.inputs.version}}-rh-ubi
+            dockerfile: dockerfile/rh-ubi/Dockerfile
+            script: build_amd64.sh
+          - image_id: factorhouse/kpow
+            image_tag: ${{github.event.inputs.version}}-temurin-ubi
+            dockerfile: dockerfile/temurin-ubi/Dockerfile
+            script: build_all.sh
+          - image_id: factorhouse/kpow
+            image_tag: ${{github.event.inputs.version}}-compat
+            dockerfile: dockerfile/compat/Dockerfile
+            script: build_all.sh
+    steps:
+      - uses: actions/checkout@v4
 
-      - name: Build Kpow
-        run: |
-          VERSION=${{github.event.inputs.version}}
-          IMAGE_ID=factorhouse/kpow
-          IMAGE_TAG=$VERSION
-          ./scripts/build_all.sh $VERSION $IMAGE_ID $IMAGE_TAG dockerfile/kpow/Dockerfile
+      - name: Download JARs
+        uses: actions/download-artifact@v4
+        with:
+          name: kpow-jars
+          path: target/
 
-      - name: Build Kpow (Java25)
-        run: |
-          VERSION=${{github.event.inputs.version}}
-          IMAGE_ID=factorhouse/kpow
-          IMAGE_TAG=$VERSION-corretto25
-          ./scripts/build_all.sh $VERSION $IMAGE_ID $IMAGE_TAG dockerfile/kpow-amazoncorretto25/Dockerfile
+      - name: Login to DockerHub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_USER }}
+          password: ${{ secrets.DOCKER_TOKEN }}
 
-      - name: Build Kpow (latest)
-        run: |
-          VERSION=${{github.event.inputs.version}}
-          IMAGE_ID=factorhouse/kpow
-          IMAGE_TAG=latest
-          ./scripts/build_all.sh $VERSION $IMAGE_ID $IMAGE_TAG dockerfile/kpow/Dockerfile
+      - name: Log into GitHub packages
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
 
-      - name: Build Kpow Red Hat UBI
-        run: |
-          VERSION=${{github.event.inputs.version}}
-          IMAGE_ID=factorhouse/kpow
-          IMAGE_TAG=$VERSION-rh-ubi
-          ./scripts/build_amd64.sh $VERSION $IMAGE_ID $IMAGE_TAG dockerfile/rh-ubi/Dockerfile
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
 
-      - name: Build Kpow Temurin UBI
+      - name: Build Docker image
         run: |
           VERSION=${{github.event.inputs.version}}
-          IMAGE_ID=factorhouse/kpow
-          IMAGE_TAG=$VERSION-temurin-ubi
-          ./scripts/build_all.sh $VERSION $IMAGE_ID $IMAGE_TAG dockerfile/temurin-ubi/Dockerfile
+          IMAGE_ID=${{ matrix.image_id }}
+          IMAGE_TAG=${{ matrix.image_tag }}
+          ./scripts/${{ matrix.script }} $VERSION $IMAGE_ID $IMAGE_TAG ${{ matrix.dockerfile }}
 
+  build-docker-ce:
```

Comment:
This can be added to the build matrix as well - 


```
          - image_id: factorhouse/kpow-ce
            image_tag: ${{github.event.inputs.version}}
            dockerfile: dockerfile/kpow-ce/Dockerfile
            script: build_all.sh
          - image_id: factorhouse/kpow-ce
            image_tag: latest
            dockerfile: dockerfile/kpow-ce/Dockerfile
            script: build_all.sh
```

`build_all.sh` essentially does what is inlined